### PR TITLE
Feature/sg 36 yolox model

### PR DESCRIPTION
**Pull request by yurkovak**
Changes:
* Split yolov5.py into yolov5_base.py and yolov5.py
* Allow overriding stem type with arch_params['stem_type'], by default it's defined by yolo_type and yolo_version
* Add yolox yamls
* Add a model construction test

The tremendous number of involved files and changes is due to:
* Some imports from yolov5 that now have to be yolov5_base;
* The renaming of a file is unfortunately displayed as a brand new file

### CODE REVIEW

**Comment by yurkovak** [comment]
For the reference, continues changes from #136

File: tests/unit_tests/yolox_unit_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                self.assertIsNotNone(output_train)
+
+                # THIS SHOULD RUN THE FORWARD AUGMENT
+                yolo_model.eval()
+                yolo_model.augmented_inference = True
+                output_augment = yolo_model(dummy_input)
+                self.assertIsNotNone(output_augment)
+
+                # MODEL IS ANCHORLESS
+                assert yolo_model._head.anchors.num_anchors == 0
```
Comment (shaydeci) [inline tests/unit_tests/yolox_unit_test.py:43]: why not add it the test suite runner ?

**Review by shaydeci — CHANGES_REQUESTED** [review]
minor inline comment

File: tests/unit_tests/yolox_unit_test.py (hunk 1 above)
Comment (shaydeci) [reply tests/unit_tests/yolox_unit_test.py:43]: Added it for you.

**Review by shaydeci — APPROVED** [review]
lgtm

**Pull request merged**